test/disassemble-chunk.test.ts

```diff
@@ -0,0 +1,89 @@
+/* Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { chance } from "jest-chance";
+import { readFileSync } from "fs";
+import { getWasmFixtures } from "./utils";
+import { WasmDisassembler } from "../src/WasmDis";
+import { BinaryReader } from "../src/WasmParser";
+
+function disassemble(data: Uint8Array): string {
+  const dis = new WasmDisassembler();
+  const parser = new BinaryReader();
+  parser.setData(data.buffer, 0, data.length);
+  return dis.disassemble(parser);
+}
+
+function disassembleInChunks(data: Uint8Array, chunkSize: number): string {
+  const dis = new WasmDisassembler();
+  const parser = new BinaryReader();
+
+  // Buffer to hold pending data.
+  var lines = [];
+  var buffer = new Uint8Array(chunkSize);
+  var pendingSize = 0;
+  var offsetInModule = 0;
+  for (var i = 0; i < data.length; ) {
+    if (chunkSize > data.length - i) {
+      chunkSize = data.length - i;
+    }
+    var bufferSize = pendingSize + chunkSize;
+    if (buffer.byteLength < bufferSize) {
+      var newBuffer = new Uint8Array(bufferSize);
+      newBuffer.set(buffer);
+      buffer = newBuffer;
+    }
+    while (pendingSize < bufferSize) {
+      buffer[pendingSize++] = data[i++];
+    }
+
+    // Setting parser buffer and signaling it's not complete.
+    var done = i == data.length;
+    parser.setData(buffer.buffer, 0, bufferSize, done);
+
+    // The disassemble will attemp to fetch the data as much as possible.
+    var finished = dis.disassembleChunk(parser, offsetInModule);
+    expect(finished).toBe(done);
+
+    var result = dis.getResult();
+    for (const line of result.lines) {
+      lines.push(line);
+    }
+
+    if (parser.position == 0) {
+      // Parser did not consume anything.
+      pendingSize = bufferSize;
+      continue;
+    }
+    // Shift the data to the beginning of the buffer.
+    var pending = parser.data.subarray(parser.position, parser.length);
+    pendingSize = pending.length;
+    buffer.set(pending);
+    offsetInModule += parser.position;
+  }
+
+  lines.push("");
+  return lines.join("\n");
+}
+
+describe("Disassembling", () => {
+  describe.each(getWasmFixtures())("%s", (fileName, filePath) => {
+    test("in chunks generates expected output", () => {
+      const data = new Uint8Array(readFileSync(filePath));
+      const chunkSize = chance.integer({ min: 1, max: data.length - 1 });
```

Comment:
It seems there is an excellent chance a regression to chunked reading will be missed by this test, until someone later reruns it while checking in something unrelated. What if you ran this with a few fixed chunk sizes like 1 and 5?

Reply:
Good point, Eric! I've removed the chance now and use 7, 256, 16k chunk sizes instead now (everything below 7 is a bad compromise wrt. test execution time).

Reply:
This means we are testing whether we can pause disassembly at approximately 1/7 of the locations? My preference if it wouldn't be terrible for test performance would be to have one fixed block size and a variable initial block size (test cases: read a block of 1 and then blocks of 7, read a block of 2 and then blocks of 7...) so we know we've tested breaking in the middle of everything, but if that's too expensive I'm happy with this.

Reply:
It looks like chunk size 1 and 5 is okayish (as you suggested initially). i think we'll need to find a better way to parallelize these tests, especially when we sync with the WebAssembly testsuite again, which has grown significantly.

Regarding the variable initial block: Not sure I understand exactly what you're proposing here. How would this look like in terms of the `disassembleInChunks` function (roughly)?